### PR TITLE
Property to not show the creation dialog

### LINKED ISSUE #214: Data Node Selector - Filter on selected scenario
## New Feature / Improvement

Filtering the Data Node selector by scenario would allow the user to see Data Nodes of the selected scenario without all the others. It can be really useful in an application. It would also create a "bridge" between the Scenario selector and Data Node selector.

Proposed syntax, for example:

`<|{data_node}|data_node_selector|scenario={scenario}|>`

This would show only Data nodes corresponding to the selected scenario ('scenario').

Taipy version:
Taipy 2.3.0

### LINKED ISSUE #280: Scenario Selector - optionally do not show form after clicking 'Add Scenario'
# Feature Improvement

Possibility to remove the form after clicking 'Add Scenario'. The scenario creation would be done with default values or through the `on_creation` callback.

# Environment

Taipy Develop 3.0

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Scenario to filter list in datanode_selector.

Resolves #280
Partly resolves #214.

### CODE REVIEW

File: frontend/taipy/src/ScenarioSelector.tsx [hunk 1]
```diff
@@ -277,7 +278,7 @@ const ScenarioEditDialog = ({ scenario, submit, open, actionEdit, configs, close
                                         <DatePicker
                                             label="Date"
                                             value={new Date(form.values.date)}
-                                            onChange={(date) => form.setFieldValue("date", date?.toISOString())}
+                                            onChange={(date?:Date|null) => form.setFieldValue("date", date?.toISOString())}
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioSelector.tsx:281]: Do we need the |null when ? is used? just asking... seems strange

File: taipy/gui_core/_context.py [hunk 2]
```diff
@@ -361,6 +361,16 @@ def crud_scenario(self, state: State, id: str, payload: t.Dict[str, str]):  # no
                         return
                 elif on_creation is not None:
                     _warn(f"on_creation(): '{on_creation}' is not a function.")
+                elif not with_dialog:
+                    if len(Config.scenarios) == 2:
+                        scenario_config = [sc for k, sc in Config.scenarios.items() if k != "default"][0]
+                    else:
+                        state.assign(
+                            _GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR,
+                            f"Error creating Scenario: need only one scenario config ({len(Config.scenarios)}).",
```
Comment (FabienLelaquais) [inline taipy/gui_core/_context.py]:
```suggestion
                            f"Error creating Scenario: only one scenario config needed ({len(Config.scenarios)} found).",
```

File: frontend/taipy/src/ScenarioSelector.tsx [hunk 3]
```diff
@@ -466,6 +444,33 @@ const ScenarioSelector = (props: ScenarioSelectorProps) => {
...
+    const onDialogOpen = useCallback(() => {
+        if (showDialog) {
+            setOpen(true);
+            setActionEdit(false);
+        } else {
+            onSubmit(false, false, {}, false);
+        }
+    }, [showDialog, onSubmit]);
+
+    const onDialogClose = useCallback(() => {
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioSelector.tsx:456]: Does this need to be a hook?

**Review by FabienLelaquais — DISMISSED** [review]
👌

File: frontend/taipy/src/ScenarioSelector.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/ScenarioSelector.tsx:281]: ? adds undefined
but the signature needs null too (when you do not select a date)

File: frontend/taipy/src/ScenarioSelector.tsx (hunk 3 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/ScenarioSelector.tsx:456]: yes because it depends on setState actions that are only available in the component

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**